### PR TITLE
add option WITH_MKL 

### LINKED ISSUE #5725: unify WITH_MKLDNN and WITH_MKLML tobe WITH_MKL
Requested by @luotao1 

We only need expose only one option `WITH_MKL` to user.

Then the `WITH_MKLML `and `WITH_MKLDNN `should be auto set by Paddle.

### PR DESCRIPTION
**Pull request by tensor-tang**
fix #5725 
Expose only one main option `WITH_MKL` to user, and hide the `WITH_MKLML` and `WITH_MKLDNN`.

### CODE REVIEW

**Comment by tensor-tang** [comment]
> [16:31:53]W:	 [Step 1/1] In file included from /paddle/paddle/operators/math/math_function.cc:16:0:
[16:31:53]W:	 [Step 1/1] /paddle/paddle/framework/data_type.h:17:43: fatal error: paddle/framework/framework.pb.h: No such file or directory
[16:31:53]W:	 [Step 1/1] compilation terminated.
[16:31:53]W:	 [Step 1/1] make[2]: *** [paddle/operators/math/CMakeFiles/math_function.dir/math_function.cc.o] Error 1
[16:31:53]W:	 [Step 1/1] make[2]: *** Waiting for unfinished jobs....


貌似出现了不是我引起的error，我重启了几遍

**Comment by luotao1** [comment]
The error is duplicated with https://github.com/PaddlePaddle/Paddle/issues/5698

File: doc/design/mkldnn/README.MD [hunk 1]
```diff
@@ -36,13 +36,13 @@ Figure 1. PaddlePaddle on IA.
 我们把集成方案大致分为了如下几个方面。
 
 ### CMake
-我们会在`CMakeLists.txt`中会添加`WITH_MKLDNN`的选项，当设置这个值为`ON`的时候会启用编译MKL-DNN功能。同时会自动开启OpenMP用于提高MKL-DNN的性能。
+我们会在`CMakeLists.txt`中会给用户添加一个`WITH_MKL`的开关，他是负责`WITH_MKLML`和`WITH_MKLDNN`的总开关。
```
Comment (luotao1) [inline doc/design/mkldnn/README.MD]: 他-》它

Comment (tensor-tang) [reply doc/design/mkldnn/README.MD]: 好的，thanks

**Pull request merged**